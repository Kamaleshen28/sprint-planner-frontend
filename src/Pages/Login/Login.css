.login-page {
  display: flex;
  color: black;
}


.login-wrapper {
  flex: 4;
  min-height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.form-container {
  width: 500px;
  height: 350px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  border-radius: 10px;
  box-shadow: 0px 0px 15px -9px #000000;
}

@media only screen and (max-width: 630px) {  
  .form-container { 
    width: 350px;
    height: 400px;

  }
  .form-input{
    width: 90%;
  }
}

.form-input {
  width: 50%;
  height: 30px;
  font-size: 20px;
  border-radius: 5px;
}
@media only screen and (max-width: 630px) {  
  .form-input{
    width: 70%;
  }
}

.form {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 30px;
  width: 100%;
}

/* form-input:focus{outline: none;} */

.no-outline:focus {
  outline: none;
}

.submit-button {
  width: 52%;
  height: 34px;
  border-radius: 5px;
  border: none;
  cursor: pointer;
  color: white;
  font-size: 20px;
  background-color: #7B7FFE;
}

.confirmation-text {
  margin-bottom: 10px;
}

.login-confirmation {
  color: black;
  font-size: 20px;
}

.register-confirmation {
  color: black;
  font-size: 20px;
}

.login-text {
  color: blue;
  cursor: pointer;
}

.register-text {
  color: blue;
  cursor: pointer;
}

.username-text {
  font-size: 20px;
  color: black;
  width: 50%;
  text-align: left;
}

.password-text {
  font-size: 20px;
  color: black;
  width: 50%;
  text-align: left;
}

@media only screen and (max-width: 630px) {  
  .username-text{
    width: 70%;
  }
  .password-text{
    width: 70%;
  }
}
.middle-login {
  display: flex;
  justify-content: center;
  width: 100%;
  height: 60%;
}

