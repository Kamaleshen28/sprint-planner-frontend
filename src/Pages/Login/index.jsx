import React, { useState } from 'react';
import './Login.css';
import axios from 'axios';
import { useNavigate } from 'react-router-dom';

export default function Login() {
  const navigate = useNavigate();
  const [haveAccount, setHaveAccount] = useState(true);
  const [formData, setFormData] = useState({
    username: '',
    password: '',
  });
  const [registerFormData, setRegisterFormData] = useState({
    username: '',
    password: '',
    confirmPassword: '',
  });

  const handleAccountConfirmationClick = () => {
    setHaveAccount((previousValue) => !previousValue);
  };

  const handleRegisterDetailChange = (event) => {
    const { name, value } = event.target;
    setRegisterFormData((previousData) => ({
      ...previousData,
      [name]: value,
    }));
  };

  const handleChange = (event) => {
    const { name, value } = event.target;
    setFormData((previousData) => ({
      ...previousData,
      [name]: value,
    }));
  };

  const handleRegisterSubmit = async (event) => {
    event.preventDefault();
    if (registerFormData.password === registerFormData.confirmPassword) {
      await callCreateUserAPI(registerFormData);
    } else {
      alert('Password does not match');
    }
  };
  const handleLoginSubmit = async (event) => {
    event.preventDefault();
    await sendLoginData(formData);
  };

  const sendLoginData = async (loginCredentials) => {
    try {
      const response = await axios.post('http://localhost:8080/auth/login', {
        username: loginCredentials.username,
        password: loginCredentials.password,
      });

      if (response.data.message === 'User logged in successfully') {
        localStorage.setItem('token', response.data.data.access_token);
        navigate('/create');
      }
    } catch (error) {
      alert(error.response.data.message);
    }
  };

  const callCreateUserAPI = async (userDetails) => {
    try {
      await axios.post('http://localhost:8080/auth/signup', {
        username: userDetails.username,
        password: userDetails.password,
      });
      alert('User created successfully');
      setHaveAccount(true);
    } catch (error) {
      alert(error.response.data.message);
    }
  };

  return (
    <div className="login-page">
      <div className="login-wrapper">
        <div className="form-container">
          {haveAccount && (
            <form className="form" onSubmit={handleLoginSubmit}>
              <span className="username-text">Username</span>
              <input
                type="text"
                className="form-input"
                value={formData.username}
                name="username"
                onChange={handleChange}
              />
              <br />
              <span className="password-text">Password</span>
              <input
                type="password"
                className="form-input"
                value={formData.password}
                name="password"
                onChange={handleChange}
              />
              <br />
              <button className="submit-button">Login</button>
            </form>
          )}
          {!haveAccount && (
            <form className="form" onSubmit={handleRegisterSubmit}>
              <span className="username-text">Username</span>

              <input
                type="text"
                className="form-input"
                value={registerFormData.username}
                name="username"
                onChange={handleRegisterDetailChange}
              />
              <br />
              <span className="password-text">Password</span>

              <input
                type="password"
                className="form-input"
                value={registerFormData.password}
                name="password"
                onChange={handleRegisterDetailChange}
              />
              <br />
              <span className="password-text">Confirm Password</span>

              <input
                type="password"
                className="form-input"
                value={registerFormData.confirmPassword}
                name="confirmPassword"
                onChange={handleRegisterDetailChange}
              />
              <br />
              <button className="submit-button">Register</button>
            </form>
          )}
          <div className="confirmation-text">
            {!haveAccount && (
              <span className="login-confirmation">
                Already have an account?{' '}
                <span
                  className="login-text"
                  onClick={handleAccountConfirmationClick}
                >
                  Login
                </span>
              </span>
            )}
            {haveAccount && (
              <span className="register-confirmation">
                Dont have an account?{' '}
                <span
                  className="register-text"
                  onClick={handleAccountConfirmationClick}
                >
                  Register
                </span>
              </span>
            )}
          </div>
        </div>
      </div>
    </div>
  );
}
