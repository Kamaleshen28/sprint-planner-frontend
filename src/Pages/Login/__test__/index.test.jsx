import React from 'react';
import { render } from '@testing-library/react';
import Login from '..';

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => jest.fn(),
  };
});

describe('Login', () => {
  it('should render correctly', () => {
    const { container } = render(<Login />);
    expect(container).toMatchSnapshot();
  });
});
