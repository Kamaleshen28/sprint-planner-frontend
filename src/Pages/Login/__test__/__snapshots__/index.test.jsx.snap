// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<div>
  <div
    class="login-page"
  >
    <div
      class="left-top-decor"
    />
    <div
      class="right-top-decor"
    />
    <div
      class="left-bottom-decor"
    />
    <div
      class="right-bottom-decor"
    />
    <div
      class="login-wrapper"
    >
      <div
        class="login-image-container"
      />
      <div
        class="form-container"
      >
        <form
          class="form"
        >
          <span
            class="username-text"
          >
            Username
          </span>
          <input
            class="form-input"
            name="username"
            type="text"
            value=""
          />
          <br />
          <span
            class="password-text"
          >
            Password
          </span>
          <input
            class="form-input"
            name="password"
            type="password"
            value=""
          />
          <br />
          <button
            class="submit-button"
          >
            Login
          </button>
        </form>
        <div
          class="confirmation-text"
        >
          <span
            class="register-confirmation"
          >
            Don't have an account?
             
            <span
              class="register-text"
            >
              Register
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
